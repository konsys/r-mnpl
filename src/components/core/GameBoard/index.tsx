import React from "react";

import { useStore } from "effector-react";

import { IFieldModalPosition } from "../../../types/types";
import { GameBoardView } from "./GameBoardView";

export const fieldsActionPosition = (): IFieldModalPosition => {
  return {
    left: 0,
    top: 0,
  };
};

export const GameBoard = () => {
  return <GameBoardView />;
};
