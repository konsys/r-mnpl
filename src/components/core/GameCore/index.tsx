import React from "react";
import { RouteComponentProps } from "@reach/router";
import "./style.scss";
import { Chat } from "../../views/Chat/Chat";
import { Contract } from "../../views/Contract/Contract";
import { TableHelper } from "../../views/TableHelper/TableHelper";
import { Tokens } from "../../views/Tokens/Tokens";
import { Ticket } from "../../views/Ticket/ticket";
import { Arbitr } from "../../views/Arbitr/Arbitr";
import { M1tv } from "../../views/M1tv/M1tv";
import { BoardModal } from "../../views/BoardModal/BoardModal";
import { Dices } from "../../views/Dices/Dices";
import { GameLoading } from "../../views/GameLoading/GameLoading";
import { useStore } from "effector-react";
import { BoardCore } from "../BoardCore/BoardCore";
import { UsersCore } from "../PlayersCore/PlayersCore";
import { actionsStore } from "../../../stores/ActionStore";
import { IncomeMessageType } from "../../../types/BoardTypes";
import "react-toastify/dist/ReactToastify.css";
import { ToastContainer } from "react-toastify";

interface Props extends RouteComponentProps {}

export const Game = (props: Props) => {
  const actionState = useStore(actionsStore);
  const actionType = actionState.event.action.type;

  const isModalAction =
    actionType === IncomeMessageType.INCOME_AUCTION_MODAL ||
    actionType === IncomeMessageType.INCOME_CAN_BUY_MODAL ||
    actionType === IncomeMessageType.INCOME_ROLL_DICES_MODAL ||
    actionType === IncomeMessageType.INCOME_TAX_PAYING_MODAL ||
    actionType === IncomeMessageType.INCOME_UN_JAIL_MODAL;

  return (
    <>
      <div className="wrapper" style={{ width: "100%", height: "100%" }}>
        <div className="table _shakehack">
          <div className="table-body">
            <UsersCore />
            <div className="table-body-board">
              <BoardCore />
              <div className="table-body-board-center">
                <M1tv />
                {actionState && (
                  // TODO check for user credentials
                  // userState.userId === actionState.event.action.userId &&
                  <BoardModal isModal={isModalAction} />
                )}
                <Arbitr />
                <Ticket />
                <Chat />
              </div>
              <div className="table-body-board-tokens">
                <Tokens />
              </div>
              <Dices />
              <Contract />
              <TableHelper />
            </div>
          </div>

          <div id="placeholder_gameover"></div>
        </div>
        <div className="table-jokes"></div>
      </div>
      <GameLoading />
      <ToastContainer />
    </>
  );
};
