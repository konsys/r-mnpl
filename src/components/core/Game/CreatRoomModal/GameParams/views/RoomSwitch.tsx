import { Grid, Switch } from "@material-ui/core";

import { GRID_SPACING } from "../../../../../../theme";
import React from "react";
import { useTranslation } from "react-i18next";

export default function RoomSwitch({
  text,
  name,
  onChange,
}: {
  text: string;
  name: string;
  onChange: (name: string) => void;
}) {
  const { t } = useTranslation();

  return (
    <Grid
      container
      justify="space-between"
      spacing={GRID_SPACING}
      alignItems="center"
    >
      <Grid item>{t(text)}</Grid>
      <Grid item>
        <Switch
          color="primary"
          checked={false}
          onChange={() => onChange(name)}
          name={name}
          inputProps={{ "aria-label": name }}
        />
      </Grid>
    </Grid>
  );
}
