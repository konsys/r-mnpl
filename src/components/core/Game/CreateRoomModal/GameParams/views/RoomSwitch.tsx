import { Grid, Switch } from "@material-ui/core";

import { GRID_SPACING } from "../../../../../../theme";
import React from "react";
import { useTranslation } from "react-i18next";

export default function RoomSwitch({
  text,
  name,
  checked,
  onChange,
  disabled,
}: {
  text: string;
  name: string;
  checked: boolean;
  onChange: (name: string) => void;
  disabled: boolean;
}) {
  const { t } = useTranslation();
  return (
    <Grid
      container
      justify="space-between"
      spacing={GRID_SPACING}
      alignItems="center"
    >
      <Grid item>{t(text)}</Grid>
      <Grid item>
        <Switch
          disabled={disabled}
          color="primary"
          checked={checked}
          onChange={() => onChange(name)}
          name={name}
          inputProps={{ "aria-label": name }}
        />
      </Grid>
    </Grid>
  );
}
