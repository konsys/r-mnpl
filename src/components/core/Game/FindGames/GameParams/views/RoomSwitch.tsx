import { Grid, Switch } from "@material-ui/core";

import { GRID_SPACING } from "../../../../../../theme";
import { IRoomSetup } from "../../CreateGameModal";
import React from "react";
import { useTranslation } from "react-i18next";

export default function RoomSwitch({
  setup,
  text,
  parameterName,
}: {
  setup: IRoomSetup | any;
  text: string;
  parameterName: any;
}) {
  const { t } = useTranslation();

  return (
    <Grid
      container
      justify="space-between"
      spacing={GRID_SPACING}
      alignItems="center"
    >
      <Grid item>{t(text)}</Grid>
      <Grid item>
        <Switch
          color="primary"
          checked={setup.state[parameterName]}
          onChange={() =>
            setup.setState({
              ...setup.state,
              [parameterName]: !setup.state[parameterName],
            })
          }
          name={parameterName}
          inputProps={{ "aria-label": parameterName }}
        />
      </Grid>
    </Grid>
  );
}
