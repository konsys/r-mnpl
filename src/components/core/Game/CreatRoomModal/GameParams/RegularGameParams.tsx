import { Grid, Typography } from "@material-ui/core";
import { newRoomStore, toggleRoomSwitch } from "stores/Game/Room/RoomStore";

import { GRID_SPACING } from "../../../../../theme";
import PlayersNumber from "./views/PlayersNumber";
import React from "react";
import RoomSwitch from "./views/RoomSwitch";
import { useStore } from "effector-react";
import { useTranslation } from "react-i18next";

export default function RegularGameParams() {
  const { t } = useTranslation();
  const room = useStore(newRoomStore);
  return (
    <Grid container direction="column" spacing={GRID_SPACING}>
      <Grid item>
        <Typography variant="h6">{t("Regular game")}</Typography>
      </Grid>
      <Grid item>
        <PlayersNumber />
      </Grid>
      <Grid item>
        <Grid container spacing={GRID_SPACING}>
          <Grid item>
            <Typography variant="h6">{t("Room settings")}</Typography>
          </Grid>
        </Grid>
        <RoomSwitch
          text={"Private room"}
          name={"privateRoom"}
          checked={room.privateRoom}
          onChange={toggleRoomSwitch}
        />
        <RoomSwitch
          text={"Game autostart"}
          name={"autostart"}
          checked={room.autostart}
          onChange={toggleRoomSwitch}
        />
      </Grid>
    </Grid>
  );
}
