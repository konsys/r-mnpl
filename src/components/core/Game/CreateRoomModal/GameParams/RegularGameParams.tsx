import { Grid, Typography } from "@material-ui/core";
import {
  preparatoryRoom$,
  toggleRoomSwitch,
} from "stores/Game/Rooms/RoomsModel";

import { GRID_SPACING } from "../../../../../theme";
import PlayersNumber from "./views/PlayersNumber";
import React from "react";
import RoomSwitch from "./views/RoomSwitch";
import { useStore } from "effector-react";
import { useTranslation } from "react-i18next";

export default function RegularGameParams({ isVip }: { isVip: boolean }) {
  const { t } = useTranslation();
  const room = useStore(preparatoryRoom$);
  return (
    <Grid container direction="column" spacing={GRID_SPACING}>
      <Grid item>
        <Typography variant="h6">{t("Regular game")}</Typography>
        <Typography variant="subtitle2" color="error">
          {!isVip && t("Open VIP status for all available options")}
        </Typography>
      </Grid>
      <Grid item>
        <PlayersNumber isVip={isVip} />
      </Grid>
      <Grid item>
        <Grid container spacing={GRID_SPACING}>
          <Grid item>
            <Typography variant="h6">{t("Room settings")}</Typography>
          </Grid>
        </Grid>
        <RoomSwitch
          text={"Private room"}
          name={"privateRoom"}
          checked={room.privateRoom}
          onChange={toggleRoomSwitch}
          disabled={!isVip}
        />
        <RoomSwitch
          text={"Game autostart"}
          name={"autostart"}
          checked={room.autostart}
          onChange={toggleRoomSwitch}
          disabled={!isVip}
        />
      </Grid>
    </Grid>
  );
}
