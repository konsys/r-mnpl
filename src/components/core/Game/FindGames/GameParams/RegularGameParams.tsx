import { Grid, Typography } from "@material-ui/core";

import { GRID_SPACING } from "../../../../../theme";
import { IRoomSetup } from "../CreateGameModal";
import PlayersNumber from "./views/PlayersNumber";
import React from "react";
import RoomSwitch from "./views/RoomSwitch";
import { useTranslation } from "react-i18next";

export default function RegularGameParams({ setup }: { setup: IRoomSetup }) {
  const { t } = useTranslation();

  return (
    <Grid container direction="column" spacing={GRID_SPACING}>
      <Grid item>
        <Typography variant="h6">{t("Regular game")}</Typography>
      </Grid>
      <Grid item>
        <PlayersNumber setup={setup} />
      </Grid>
      <Grid item>
        <Grid container spacing={GRID_SPACING}>
          <Grid item>
            <Typography variant="h6">{t("Room settings")}</Typography>
          </Grid>
        </Grid>
        <RoomSwitch
          setup={setup}
          text={"Private room"}
          parameterName={"privateRoom"}
        />
        <RoomSwitch
          setup={setup}
          text={"Game autostart"}
          parameterName={"autostart"}
        />
      </Grid>
    </Grid>
  );
}
