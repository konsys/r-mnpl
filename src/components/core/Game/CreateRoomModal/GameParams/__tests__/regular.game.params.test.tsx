import React from "react";
import RegularGameParams from "../RegularGameParams";
import RoomSwitch from "../views/RoomSwitch";
import { createShallow } from "@material-ui/core/test-utils";
import { shallow } from "enzyme";

describe("Test regular geame params", () => {
  let shall: typeof shallow;

  beforeAll(() => {
    shall = createShallow();
  });

  afterAll(() => {
    // @ts-ignore
    shall.cleanUp();
  });

  it("should render", () => {
    expect(shallow(<RegularGameParams isVip />)).toMatchSnapshot();
  });

  it("should render 2 switches", () => {
    expect(shall(<RegularGameParams isVip />).find(RoomSwitch)).toHaveLength(2);
  });

  it("should render RoomSwitch vip", () => {
    shall(<RegularGameParams isVip />)
      .find(RoomSwitch)
      .forEach((v) => expect(v.prop("disabled")).toBeFalsy());
  });

  it("should render RoomSwitch vip", () => {
    shall(<RegularGameParams isVip={false} />)
      .find(RoomSwitch)
      .forEach((v) => expect(v.prop("disabled")).toBeTruthy());
  });
});
