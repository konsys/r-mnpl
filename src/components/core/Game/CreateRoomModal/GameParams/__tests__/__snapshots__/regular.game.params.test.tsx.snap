// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test regular geame params should render 1`] = `ShallowWrapper {}`;
